### ISSUE TITLE
Payload doesn't work with pnpm while making use of `useFieldType`

### ISSUE THREAD

**Issue by lachieh** [payloadcms/payload#665]
# Bug Report

While getting set up with Payload for the first time I used `pnpm` to install the project. Everything was going swimmingly until I started to create a custom form field component and used the `useFieldType` hook from `payload/components/forms`. The error that came up pointed me to this issue: #416

## Reproduction Steps
1. Clone the existing [repro repo](https://github.com/payloadcms/issue-416) (that's fun to say)
2. use [pnpm](https://pnpm.io/) to install the packages:
    `pnpm i`
3. run the dev server:
    `pnpm run dev`
    ```
    /Volumes/SSD/Projects/payload-poc/node_modules/.pnpm/payload@0.18.0_3u6mezahkkqoy6xqszupxyezbi/node_modules/payload/components/forms.ts:1
    payload-poc:dev: export {
    payload-poc:dev: ^^^^^^
    payload-poc:dev: 
    payload-poc:dev: SyntaxError: Unexpected token 'export'
    payload-poc:dev:     at compileFunction (<anonymous>)
    payload-poc:dev:     at Object.compileFunction (node:vm:352:18)
    payload-poc:dev:     at wrapSafe (node:internal/modules/cjs/loader:1032:15)
    payload-poc:dev:     at Module._compile (node:internal/modules/cjs/loader:1067:27)
    payload-poc:dev:     at Module._compile (/Volumes/SSD/Projects/payload-poc/node_modules/.pnpm/pirates@4.0.5/node_modules/pirates/lib/index.js:136:24)
    payload-poc:dev:     at Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    payload-poc:dev:     at require.extensions.<computed> (/Volumes/SSD/Projects/payload-poc/node_modules/.pnpm/ts-node@10.8.1_zag423k5mmeq7scacelqeecc34/node_modules/ts-node/src/index.ts:1587:43)
    payload-poc:dev:     at Object.newLoader [as .ts] (/Volumes/SSD/Projects/payload-poc/node_modules/.pnpm/pirates@4.0.5/node_modules/pirates/lib/index.js:141:7)
    payload-poc:dev:     at Module.load (node:internal/modules/cjs/loader:981:32)
    payload-poc:dev:     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    ```

## Possible Solution
The detailed breakdown of the problem in #416 really helped me understand the process, so thank you @jmikrut!

The issue lies with the way that pnpm handles the actual file structure for `node_modules`. It symlinks everything into its own structure and therefore the path is not the same as a standard npm/yarn installation.

Here is an example of the file path created by pnpm:

`/project/node_modules/.pnpm/payload@0.18.0_3u6mezahkkqoy6xqszupxyezbi/node_modules/payload/components/forms.ts`

I was able to fix the issue by changing the RegExp pattern here:
https://github.com/payloadcms/payload/blob/70edb0ba38a1a96cefafb0edb03b06add0a86eef/src/config/load.ts#L32
to this:
```diff
    ignore: [
-      /node_modules[\\/](?!payload[\\/]dist[\\/]admin|payload[\\/]components).*/,
+      /node_modules[\\/](?!.pnpm[\\/].*[\\/]node_modules[\\/])(?!payload[\\/]dist[\\/]admin|payload[\\/]components).*/, 
    ],
```

I usually use regexr.com to plan/test complex patterns, so here is the link I used:
https://regexr.com/6nuep

Though it's a small change, I'd love to contribute this with a PR if the team is willing to accept one!

**Comment by EasonSoong** [comment]
same issue +1

**Comment by lachieh** [comment]
Since I'm not the only one to discover this, I've created the PR.

**Comment by jmikrut** [comment]
Thanks @lachieh for the PR! Fix deployed in `0.18.4`.

@EasonSoong tagging you for visibility.
